Networking/Networking/Model/CouponReport.swift

```diff
@@ -0,0 +1,46 @@
+import Codegen
+import Foundation
+
+public struct CouponReport {
+    /// ID of the coupon
+    public let couponId: Int64
+
+    /// Total amount deducted from orders using the coupon
+    public let amount: Double
+
+    /// Total number of orders that used the coupon
+    public let ordersCount: Int64
+
+    public init(couponId: Int64,
+                amount: Double,
+                ordersCount: Int64) {
+        self.couponId = couponId
+        self.amount = amount
+        self.ordersCount = ordersCount
+    }
+}
+
+// MARK: - Decodable Conformance
+//
+extension CouponReport: Decodable {
+    /// Defines all of the CouponReport CodingKeys
+    /// The model is intended to be decoded with`JSONDecoder.KeyDecodingStrategy.convertFromSnakeCase`
+    /// so any specific `CodingKeys` provided here should be in camel case.
+    enum CodingKeys: String, CodingKey {
+        case couponId, amount, ordersCount
```

Comment:
nit: same on the `*ID`

```suggestion
        case couponID, amount, ordersCount
```